addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudConnection.java

```diff
@@ -58,31 +57,30 @@ public ICloudConnection(String appleId, String password)
      * @throws IOException
      */
     public void findMyDevice(String id) throws IOException {
-        String iCloudFindMyDeviceRequest = gson.toJson(new ICloudFindMyDeviceRequest(id));
-        HttpUtil.executeUrl("POST", iCloudFindMyDeviceURL.toString(), httpHeader,
-                new ByteArrayInputStream(iCloudFindMyDeviceRequest.getBytes("UTF-8")), "application/json",
-                SOCKET_TIMEOUT);
+        callApi(iCloudFindMyDeviceURL, gson.toJson(new ICloudFindMyDeviceRequest(id)));
     }
 
     public String requestDeviceStatusJSON() throws IOException {
-        return HttpUtil.executeUrl("POST", iCloudDataRequestURL.toString(), httpHeader,
-                new ByteArrayInputStream(iCloudDataRequest.getBytes("UTF-8")), "application/json", SOCKET_TIMEOUT);
+        return callApi(iCloudDataRequestURL, iCloudDataRequest);
     }
 
-    private Properties createHttpHeader(byte[] authorization) throws UnsupportedEncodingException {
-        Properties httpHeader = new Properties();
-
-        httpHeader.setProperty("Authorization", "Basic " + new String(authorization, "UTF-8"));
-        httpHeader.setProperty("User-Agent", "Find iPhone/1.3 MeKit (iPad: iPhone OS/4.2.1)");
-        httpHeader.setProperty("Origin", ICLOUD_URL);
-        httpHeader.setProperty("charset", "utf-8");
-        httpHeader.setProperty("Accept-language", "en-us");
-        httpHeader.setProperty("Connection", "keep-alive");
-        httpHeader.setProperty("X-Apple-Find-Api-Ver", "2.0");
-        httpHeader.setProperty("X-Apple-Authscheme", "UserIdGuest");
-        httpHeader.setProperty("X-Apple-Realm-Support", "1.0");
-        httpHeader.setProperty("X-Client-Name", "iPad");
-
-        return httpHeader;
+    private String callApi(String url, String payload) throws IOException {
+        // @formatter:off
+        return HttpRequestBuilder.postTo(url)
+            .withTimeout(Duration.ofSeconds(SOCKET_TIMEOUT))
+            .withHeader("Authorization", "Basic " + authorization)
+            .withHeader("User-Agent", "Find iPhone/1.3 MeKit (iPad: iPhone OS/4.2.1)")
+            .withHeader("Origin", ICLOUD_URL)
+            .withHeader("charset", "utf-8")
+            .withHeader("Accept-language", "en-us")
+            .withHeader("Connection", "keep-alive")
+            .withHeader("X-Apple-Find-Api-Ver", "2.0")
+            .withHeader("X-Apple-Authscheme", "UserIdGuest")
+            .withHeader("X-Apple-Realm-Support", "1.0")
+            .withHeader("X-Client-Name", "iPad")
+            .withHeader("Content-Type", "application/json")
```

Comment:
This seems to be a potential additional header but the rest seems to have been there already. Did it work flawlessly before they changed the certificate or could it be something related to the way the password is transferred combined with the length of your username password?

My account was online today, but I recently removed a device from my account and all my devices reported OFFLINE. After I removed the device which I do not own anymore and disabled/enabled the account all reported fine again.

The logging which I had was `Unable to find device data.` Looking a the code that does not sound like a reason it would not work, maybe adding some more information / logging around: https://github.com/openhab/openhab2-addons/blob/ee2d4a5159fbe8dcb5c0518078c74209858ed406/addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/handler/ICloudDeviceHandler.java#L77 would help.

I think your issue should go to a new issue because I don't think it's directly caused by the changes from this PR.

Reply:
Just updated to the latest snapshot and now my account is online. I hence cannot reproduce it myself, so let's forget about it for now :-)